src/Serilog.Sinks.RollingFile/Sinks/RollingFile/TemplatedPathRoller.cs

```diff
@@ -29,18 +29,42 @@ class TemplatedPathRoller
         const string OldStyleDateSpecifier = "{0}";
         const string DateSpecifier = "{Date}";
         const string DateFormat = "yyyyMMdd";
+        const string HourSpecifier = "{Hour}";
+        const string HourFormat = "yyyyMMddHH";
+        const string HalfHourSpecifier = "{HalfHour}";
+        const string HalfHourFormat = "yyyyMMddHHmm";
         const string DefaultSeparator = "-";
 
+        const string MatcherMarkSpecifier = "date";
+        const string MatcherMarkInc = "inc";
+
         readonly string _pathTemplate;
         readonly Regex _filenameMatcher;
+        readonly SpecifierTypeEnum _specifierType = SpecifierTypeEnum.None;
```

Comment:
For me it is great to encapsulate the fields in a class. 
I didn't want to make so much changes from the original source so the merge should be easier.
Feel free to make any change with which you feel more confortable.